api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/Eri2MonthsCohortQueries.java

```diff
@@ -0,0 +1,338 @@
+        EptsReportUtils.map(
+            getAllPatientsWhoInitiatedArt(), "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "pregnant",
+        EptsReportUtils.map(
+            txNewCohortQueries.getPatientsPregnantEnrolledOnART(),
+            "startDate=${endDate-2m},endDate=${endDate-1m},location=${location}"));
+    cd.setCompositionString("all AND pregnant");
+    return cd;
+  }
+
+  /**
+   * Get breastfeeding women who have more than 2 months ART retention
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getBreastfeedingWomenRetainedOnArtFor2MonthsFromArtInitiation() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("Breastfeeding women retain on ART for more than 2 months from ART initiation date");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location" + "", "Location", Location.class));
+    cd.addSearch(
+        "all",
+        EptsReportUtils.map(
+            getAllPatientsWhoInitiatedArt(), "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "breastfeeding",
+        EptsReportUtils.map(
+            txNewCohortQueries.getTxNewBreastfeedingComposition(),
+            "onOrAfter=${endDate-2m},onOrBefore=${endDate-1m},location=${location}"));
+    cd.setCompositionString("all AND breastfeeding");
+    return cd;
+  }
+
+  /**
+   * Get Children (0-14, excluding pregnant and breastfeeding women)
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getChildrenRetainedOnArtFor2MonthsFromArtInitiation() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName("Children having ART retention for than 2 months");
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location" + "", "Location", Location.class));
+    cd.addSearch(
+        "all",
+        EptsReportUtils.map(
+            getAllPatientsWhoInitiatedArt(), "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "children",
+        EptsReportUtils.map(
+            txPvlsCohortQueries.findPatientsBetweenAgeBracketsInYears(0, 14),
+            "endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "pregnant",
+        EptsReportUtils.map(
+            getPregnantWomenRetainedOnArtFor2MonthsFromArtInitiation(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "breastfeeding",
+        EptsReportUtils.map(
+            getBreastfeedingWomenRetainedOnArtFor2MonthsFromArtInitiation(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.setCompositionString("(all AND children) AND NOT (pregnant OR breastfeeding)");
```

Comment:
We could maybe improve this composition because currently this: `(all AND children) AND NOT (pregnant OR breastfeeding)`

is actually: (all AND children) AND NOT ((all and pregnant) OR (all and breastfeeding))`